ProductService/src/main/java/com/metaphorce/exam/ecommerce/product/controller/InventoryController.java

```diff
@@ -0,0 +1,138 @@
+import lombok.extern.slf4j.Slf4j;
+
+@RequestMapping("/inventory")
+@RestController
+@Slf4j
+public class InventoryController {
+	@Autowired
+	private InventoryService service;
+	
+	/**
+	 * Implements the HTTP Post to create a Inventory
+	 * @param Inventory Data to insert
+	 * @return New Inventory create
+	 * @author Juan Chavez
+	 * @since Jun/11/2024 
+	 */
+	@PostMapping
+	public ResponseEntity<InventoryDTO> createInventory(@Valid @RequestBody InventoryDTO InventoryRequest) {
+		log.info("Inventory to create: " + InventoryRequest);
+		ResponseEntity<InventoryDTO> response = null;
+		Inventory newInventory = this.service.create(InventoryMapper.mapper.toEntity(InventoryRequest));
+
+		if(newInventory != null) {
+			response = ResponseEntity.ok(InventoryMapper.mapper.toDTO(newInventory));
+			log.info("Inventory created: " + newInventory);
+		} else {
+			response = ResponseEntity.badRequest().build();
+			log.info("Error to create Inventory: " + InventoryRequest);
+		}
+
+		return response;
+	}
+
+	/**
+	 * Implements the HTTP Get to find all Inventorys
+	 * @return List of Inventorys
+	 * @author Juan Chavez
+	 * @since Jun/11/2024 
+	 */
+	@GetMapping
+	public ResponseEntity<List<InventoryDTO>> getAllInventories() {
+		log.info("Get All Inventorys ...");
+		ResponseEntity<List<InventoryDTO>> response = null;
+		List<Inventory> listInventorys = this.service.getAll();
+
+		if(listInventorys != null) {
+			log.info("Inventorys[" + listInventorys.size() + "]");
+			response = ResponseEntity.ok(listInventorys.stream().map(InventoryMapper.mapper::toDTO).toList());
+		} else {
+			response = ResponseEntity.notFound().build();
+			log.info("Error to get all Inventorys");
+		}
+
+		return response;
+	}
+
+	/**
+	 * Implements HTTP Get to find a Inventory by id
+	 * @param id Inventory Id
+	 * @return Inventory data
+	 * @return New Inventory create
+	 * @author Juan Chavez
+	 * @since Jun/11/2024
+	 */
+	@GetMapping("/{id}")
+	public ResponseEntity<InventoryDTO> getInventory(@PathVariable Integer id) {
+		log.info("Get Inventory[" + id + "]");
+		return service.get(id).map((Inventory Inventory) -> {
+			log.info("Inventory found: " + Inventory);
+			return ResponseEntity.ok(InventoryMapper.mapper.toDTO(Inventory));
+		}).orElseGet(() -> {
+			log.warn("Inventory not found: " + id);
+			return ResponseEntity.notFound().build();
+		});
+	}
+
+	/**
+	 * Implements HTTP Put for update a Inventory
+	 * @param id Inventory Id
+	 * @param Inventory Inventory data to update
+	 * @return Inventory modified
+	 * @return New Inventory create
+	 * @author Juan Chavez
+	 * @since Jun/11/2024
+	 */
+	@PutMapping("/{id}")
+	public ResponseEntity<InventoryDTO> updateInventory(@PathVariable Integer id, @Valid @RequestBody InventoryDTO Inventory) {
+		log.info("Inventory[" + id + "] to update: " + Inventory);
+		ResponseEntity<InventoryDTO> response = null;
+		Inventory updateInventoryData = this.service.update(id, InventoryMapper.mapper.toEntity(Inventory));
+
```

Comment:
I don't understand why check if the value is null. It would be better to throw exceptions in the service when it doesn't find a certain resource; also, add a global exception handler.

In the case of not being able to create a resource, I think it's better to let a 500 status be generated because it probably means there's something wrong with the system.